pre-shutdown-hook

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+if [ $# -eq 0 ]
+then
+    echo "usage: $0 <shutdown-deferrer url>"
+    exit 1
+fi
+
+# In order to guarantee correct behavior, this poll_interval should be equal to
+# one defined in k8s-kvm qemu-shutdown script.
+poll_interval=5
+poll_timeout=120
+shutdown_deferrer_url=$1
+
+# Poll shutdown-deferrer service in order to wait for proper node draining
+# before shutdown.
+defer="true"
+while [ "$defer" = "true" -a $poll_timeout -gt 0 ]
+do
+    sleep $poll_interval
+    defer=$(/usr/bin/curl -qsS $shutdown_deferrer_url)
+    echo "GET /v1/defer: $defer"
+    poll_timeout=$(/usr/bin/expr $poll_timeout - $poll_interval)
+done
+
+# Small delay before exit so that k8s-kvm preStop hook has chance to do last
+# defer query.
+sleep $(/usr/bin/expr $poll_interval + 1)
```

Comment:
I do not think there is a reliable way to tell that the other container's hook has really more or less time to do anything.